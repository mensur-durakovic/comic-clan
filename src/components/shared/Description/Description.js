import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const DescriptionWrapper = styled.div`
  font-size: 16px;
  line-height: 19px;
  margin-bottom: 10px;
  max-width: fit-content;
`;

const DescriptionTitle = styled.span`
  text-transform: capitalize;
  font-weight: 500;
  color: #999999;
`;

const DescriptionValue = styled.span`
  font-weight: bold;
  color: #cccccc;
`;

const Description = React.memo((props) => {
  const { descriptionTitle, descriptionValue } = props;
  return (
    <DescriptionWrapper>
      <DescriptionTitle>
        {descriptionTitle}
        {": "}
      </DescriptionTitle>
      <DescriptionValue> {descriptionValue} </DescriptionValue>
    </DescriptionWrapper>
  );
});

Description.propTypes = {
  descriptionTitle: PropTypes.string.isRequired,
  descriptionValue: PropTypes.string.isRequired,
};

export default Description;
