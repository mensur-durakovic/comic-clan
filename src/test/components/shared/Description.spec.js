import React from 'react';
import renderer from 'react-test-renderer';
import Description from '../../../components/shared/Description/Description';

test('Description component', () => {
  const tree = renderer.create(<Description descriptionTitle="test" descriptionValue="test"/>).toJSON()
  expect(tree).toMatchSnapshot()
})