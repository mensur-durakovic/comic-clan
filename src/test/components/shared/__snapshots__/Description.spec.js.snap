// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description component 1`] = `
<div
  className="sc-AxjAm fiYXvS"
>
  <span
    className="sc-AxirZ gnUxoK"
  >
    test
    : 
  </span>
  <span
    className="sc-AxiKw enTNPh"
  >
     
    test
     
  </span>
</div>
`;
